pdudaemon/drivers/cleware.py

```diff
@@ -104,7 +143,7 @@ def port_interaction(self, command, port_number):
             return
 
         d = hid.device()
-        d.open(CLEWARE_VID, CLEWARE_CONTACT00_PID, serial_number=self.serial)
+        d.open(CLEWARE_VID, CLEWARE_CONTACT00_PID, serial_number=self.find_hid_serial())
```

Comment:
I assume this is to force the serial number to be treated as a hex encoded int rather then real string;  Potentially also best then to change the example configuration to a number rather then a string in `share/pdudaemon.conf` ?

Reply:
The reason why I introduced a dedicated method for finding a serial number match, is that I wasn't sure which serial number hex format is registered (as string) in the HID device.

For example on the sticker of the HID device it says `7105`. If you convert this to hex in python, you'll get a string `'0x1bc1'`. However, `hid.enumerate()` (in my case) returns a dict with the serial number string `'0x0001BC1'`. It is obvious that these are the same, but you won't be able to find this out by comparing the strings.

That's why I introduced `find_hid_serial()`, that overcomes this, by converting back to int for the comparison.
Note that this still returns the serial number as string.

If you have a better proposal to overcome this, please let me know.

I propose to keep the string in the configuration file, for reasons of consistency. I will update to make sure `self.serial` is an integer.

Reply:
This seems an odd middleground to me; The configuration key being a string but then enforcing it everywhere to be an integer seems surprising.. Not sure what you think you're being consistent with either as these are all device specific settings

Reply:
I was thinking of older configurations, people already have in place.
But I'm wrong there, since where switching to the hex based serial number anyway.

Thus, I also will update `share/pdudaemon.conf` and change to integers there.